### DISCUSSION TITLE
Use cases for second-level precision in `UtcOffset`

### DISCUSSION THREAD

**Discussion by jhpratt** [time-rs/time#338] [Seeking input]
This discussion is for collecting use cases of having second-level precision in the `UtcOffset` struct. Specifically, I am investigating whether to move `UtcOffset` to only support hours and minutes (possibly, but not likely, restricted to quarter-hour offsets). While I have not checked the time zone database, I'm fairly certain that every location has had UTC offsets that only require minute-level precision for a significant amount of time. Removal of second-level precision will not occur in time 0.3, so there's no need to worry about this being ripped out from under your feet.

This capability may be removed in a future release if there is insufficient demand. Leave your use case below, and use the :+1: reaction if you share this use case. Please do not :+1: a use case that you do not actually need this capability for. Off-topic comments will likely be outright deleted.

**Comment by emkw** [comment]
I gone through https://data.iana.org/time-zones/releases/tzdata2021a.tar.gz and at the beginning of XX century, or even until 1972 there were cases of second-level offset, so it may be required for tzdb compatibility.

Some examples:

```
# Zone  NAME            STDOFF  RULES   FORMAT  [UNTIL]
Zone    Europe/Dublin   -0:25:00 -      LMT     1880 Aug  2
                        -0:25:21 -      DMT     1916 May 21  2:00s
                        -0:25:21 1:00   IST     1916 Oct  1  2:00s
```

```
# Zone  NAME            STDOFF  RULES   FORMAT  [UNTIL]
Zone    Africa/Monrovia -0:43:08 -      LMT     1882
                        -0:43:08 -      MMT     1919 Mar # Monrovia Mean Time
                        -0:44:30 -      MMT     1972 Jan 7 # approximately MMT
```

```
# Zone  NAME            STDOFF  RULES   FORMAT  [UNTIL]
Zone    Asia/Bangkok    6:42:04 -       LMT     1880
                        6:42:04 -       BMT     1920 Apr # Bangkok Mean Time
```

**Reply by jhpratt** [reply]
Yeah, I'm not particularly concerned about dates over 100 years ago. Monrovia is an interesting case that's recent enough to be sufficient reason imo. I'll look into this more for sure.

**Reply by emkw** [reply]
Here's full excerpt from `africa` file

```
# Liberia
#
# From Paul Eggert (2017-03-02):
#
# The Nautical Almanac for the Year 1970, p 264, is the source for -0:44:30.
#
# In 1972 Liberia was the last country to switch from a UT offset
# that was not a multiple of 15 or 20 minutes.  The 1972 change was on
# 1972-01-07, according to an entry dated 1972-01-04 on p 330 of:
# Presidential Papers: First year of the administration of
# President William R. Tolbert, Jr., July 23, 1971-July 31, 1972.
# Monrovia: Executive Mansion.
#
# Use the abbreviation "MMT" before 1972, as the more-accurate numeric
# abbreviation "-004430" would be one byte over the POSIX limit.
#
# Zone  NAME            STDOFF  RULES   FORMAT  [UNTIL]
Zone    Africa/Monrovia -0:43:08 -      LMT     1882
                        -0:43:08 -      MMT     1919 Mar # Monrovia Mean Time
                        -0:44:30 -      MMT     1972 Jan 7 # approximately MMT
                         0:00   -       GMT
```

**Reply by jhpratt** [reply]
Thanks! I'm at least familiar with tzdb so I know how to read the files. I'll likely dig into this at some point to determine if it's _actually_ what was used or if -0:44 would be sufficient. Can't hurt to ask on the mailing list if I'm not sure either.

**Reply by jhpratt** [reply]
Being unable to find any information that's freely available online, I'm going to have to say this is sufficient reason to keep the seconds field. I saw that there was the source document in some libraries, but the nearest one that has it is over four hours from me :/
